dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/ClientCore.cs

```diff
@@ -968,39 +968,71 @@ private static ChatRequestMessage GetRequestMessage(ChatRole chatRole, string co
         throw new NotImplementedException($"Role {chatRole} is not implemented");
     }
 
-    private static ChatRequestMessage GetRequestMessage(ChatMessageContent message)
+    private static IEnumerable<ChatRequestMessage> GetRequestMessages(ChatMessageContent message, ToolCallBehavior? toolCallBehavior)
     {
         if (message.Role == AuthorRole.System)
         {
-            return new ChatRequestSystemMessage(message.Content) { Name = message.AuthorName };
+            return new[] { new ChatRequestSystemMessage(message.Content) { Name = message.AuthorName } };
         }
 
-        if (message.Role == AuthorRole.User || message.Role == AuthorRole.Tool)
+        if (message.Role == AuthorRole.Tool)
         {
+            // Handling function call results represented by the TextContent type.
+            // Example: new ChatMessageContent(AuthorRole.Tool, content, metadata: new Dictionary<string, object?>(1) { { OpenAIChatMessageContent.ToolIdProperty, toolCall.Id } })
             if (message.Metadata?.TryGetValue(OpenAIChatMessageContent.ToolIdProperty, out object? toolId) is true &&
                 toolId?.ToString() is string toolIdString)
             {
-                return new ChatRequestToolMessage(message.Content, toolIdString);
+                return new[] { new ChatRequestToolMessage(message.Content, toolIdString) };
             }
 
+            // Handling function call results represented by the FunctionResultContent type.
+            // Example: new ChatMessageContent(AuthorRole.Tool, items: new ChatMessageContentItemCollection { new FunctionResultContent(functionCall, result) })
+            if (message.Items.OfType<FunctionResultContent>() is { } resultContents && resultContents.Any())
```

Comment:
OfType will never return null

Reply:
@stephentoub , I wonder if there's a more optimal way to avoid multiple enumeration of the `resultContents` variable of type IEnumerable<FunctionResultContent>? Or, is using ToArray acceptable/optimal enough in this case?
<img width="569" alt="image" src="https://github.com/microsoft/semantic-kernel/assets/68852919/8edbca24-26f5-4fb7-bd23-9aafb9a99827">

Reply:
It'd be more efficient to do something like:
```C#
List<ChatRequestToolMessage>? toolMessages = null;
foreach (var resultContent in message.Items)
{
    if (resultContent is not FunctionResultContent frc) continue;

    toolMessages ??= new();
    ... // process resultContent
}

if (toolMessages is not null)
{
    return tooMessages;
}
...
```

Reply:
Thanks, fixed.